### PR TITLE
raise error if mcmc observes subsample site

### PR DESCRIPTION
**Pull request by fehiepsi**
Addresses #2394.

### CODE REVIEW

File: pyro/infer/mcmc/util.py [hunk 1]
```diff
@@ -384,6 +384,8 @@ def initialize_model(model, model_args=(), model_kwargs={}, transforms=None, max
     prototype_samples = {}
     for name, node in model_trace.iter_stochastic_nodes():
         if isinstance(node["fn"], _Subsample):
+            if node["fn"].subsample_size:
```
Comment (fritzo) [inline pyro/infer/mcmc/util.py:387]: I think we can be more permissive and check:
```py
fn = node["fn"]
if fn.subsample_size is not None or fn.subsample_size < fn.size:
    raise ...
```
This allows passing `subsample_size=len(data)` for models that support either MCMC or SVI.

File: tests/infer/mcmc/test_mcmc_util.py [hunk 2]
```diff
@@ -68,3 +68,14 @@ def test_model_with_param(jit_compile, num_chains):
     kernel = NUTS(model_with_param, jit_compile=jit_compile, ignore_jit_warnings=True)
     mcmc = MCMC(kernel, 10, num_chains=num_chains, mp_context="spawn")
     mcmc.run()
+
+
+def test_model_with_subsample():
+    def model():
+        with pyro.plate("J", 10, subsample_size=5):
```
Comment (fritzo) [inline tests/infer/mcmc/test_mcmc_util.py:75]: we could add a test that `pyro.plate("J", 10, subsample_size=10)` works without error.

**Comment by neerajprad** [comment]
Not sure why we are seeing two travis PR checks and one is stalled. I'll reopen this PR.

**Comment by fehiepsi** [comment]
Opss... I'll make a new PR instead.

**Pull request closed without merging**